osu.Game/Screens/SelectV2/WedgeHitStatisticsDisplay.cs

```diff
@@ -0,0 +1,159 @@
+using osu.Game.Beatmaps;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Overlays;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Screens.SelectV2
+{
+    public partial class WedgeHitStatisticsDisplay : CompositeDrawable, IHasAccentColour
+    {
+        private const float statistic_width = 75;
+
+        private readonly FillFlowContainer<WedgeStatisticDifficulty> statisticsFlow;
+        private readonly GridContainer tinyStatisticsGrid;
+
+        private IReadOnlyList<BeatmapStatistic>? statistics;
+
+        public IReadOnlyList<BeatmapStatistic>? Statistics
+        {
+            get => statistics;
+            set
+            {
+                statistics = value;
+                updateStatistics();
+                updateTinyStatistics();
+            }
+        }
+
+        private Color4 accentColour;
+
+        public Color4 AccentColour
+        {
+            get => accentColour;
+            set
+            {
+                accentColour = value;
+
+                foreach (var statistic in statisticsFlow)
+                    statistic.AccentColour = value;
+            }
+        }
+
+        private readonly LayoutValue drawSizeLayout = new LayoutValue(Invalidation.DrawSize);
+
+        [Resolved]
+        private OverlayColourProvider colourProvider { get; set; } = null!;
+
+        public WedgeHitStatisticsDisplay()
+        {
+            AutoSizeAxes = Axes.Y;
+
+            InternalChildren = new Drawable[]
+            {
+                statisticsFlow = new FillFlowContainer<WedgeStatisticDifficulty>
+                {
+                    AutoSizeAxes = Axes.Both,
+                    Spacing = new Vector2(8f, 0f),
+                    Direction = FillDirection.Horizontal,
+                },
+                tinyStatisticsGrid = new GridContainer
+                {
+                    Alpha = 0f,
+                    AutoSizeAxes = Axes.Both,
+                    ColumnDimensions = new[]
+                    {
+                        new Dimension(GridSizeMode.AutoSize),
+                        new Dimension(GridSizeMode.Absolute, 8),
+                        new Dimension(GridSizeMode.AutoSize),
+                    }
+                },
+            };
+
+            AddLayout(drawSizeLayout);
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            if (!drawSizeLayout.IsValid)
+            {
+                updateLayout();
+                drawSizeLayout.Validate();
+            }
+        }
+
+        private void updateLayout()
+        {
+            if (statistics == null)
+                return;
+
+            float statisticsFlowWidth = statistic_width * statistics.Count + 8 * (statistics.Count - 1);
+            bool displayTinyStatistics = DrawWidth < statisticsFlowWidth;
+            statisticsFlow.Alpha = displayTinyStatistics ? 0 : 1;
+            tinyStatisticsGrid.Alpha = displayTinyStatistics ? 1 : 0;
```

Comment:
Was hoping for some animation, but that can be done as a follow-up as it only affects users who regularly change ui scale.

Reply:
I've added basic form of animation in https://github.com/ppy/osu/commit/6ca36855e46459d0f06a29fc144465eb0d883f41, the previous flow intentionally hides instantly to avoid showing one frame of the new statistics exceeding boundaries.